### DISCUSSION TITLE
MKR 1310 Setup

### DISCUSSION THREAD

**Discussion by engeen-nl** [jgromes/RadioLib#343] [Q&A] [answered]
I'm attempting to use the on-board SX1276 on the MKR1310 for FSK modulation. When I try the SX127x_FSK_Modem example I'm getting a
`[SX1276] Initiating ... failed, code -2`
-2 means ERR_CHIP_NOT_FOUND
I'm sure using the right modem (SX1276 address 0x12) and it is wired, because when I use sandeepmistry's library I can communicate between arduino's using LoRa modulation.
Selected pins:
`SX1276 radio = new Module(LORA_IRQ_DUMB, LORA_IRA, LORA_RESET, RADIOLIB__NC, SPI1);`
I also tried LORA_BOOT0 but no succes. I also tried the reset mechanism as used by the LoRa library of sandeepmistry, but this also doesn't work. Anyone any ideas? Help would be greatly appreciated.

**Comment by jgromes** [comment]
MKR WAN 1310 is not officially supported by the library, though it looks like it still manages to compile on the platform (somehow).

I'm not familiar with the architecture of the board, so I can't answer if it's even possible to get it running. I know it's using some Muarata module instead of plain SX1276 connected directly via SPI.

**Reply by engeen-nl** [reply]
I saw you marked the problem as a duplicate of another problem. Can you please link to the duplicated problem? Then we can look for answers or post solutions there.

**Reply by jgromes** [reply]
You have posted both in Issues, as well as in Discussions, that's why the issue was marked as duplicate.

**Comment by engeen-nl** [comment, marked as answer]
So, it seems the default SPI frequency is 2 MHz, which doesn't work on the MKR 1310. Adding `SPISettings` with a Frequency of 200 kHz and doing a `SPI1.begin()` before `radio.beginFSK` works:
```
SX1276 radio = new Module(LORA_IRQ_DUMB, LORA_IRQ, RADIOLIB_NC, RADIOLIB_NC, SPI1, SPISettings(200000, MSBFIRST, SPI_MODE0));

void setup() {
  // Reset Modem (thank you sandeepmistry and associates)
  pinMode(LORA_IRQ_DUMB, OUTPUT);
  digitalWrite(LORA_IRQ_DUMB, LOW);

  // Hardware reset
  pinMode(LORA_BOOT0, OUTPUT);
  digitalWrite(LORA_BOOT0, LOW);

  pinMode(LORA_RESET, OUTPUT);
  digitalWrite(LORA_RESET, HIGH);
  delay(200);
  digitalWrite(LORA_RESET, LOW);
  delay(200);
  digitalWrite(LORA_RESET, HIGH);
  delay(50);

  // start SPI
  SPI1.begin();
  
  // initialize SX1262 FSK modem with default settings and 869.5 MHz
  Serial.print(F("[SX1276] Initializing ... "));
  int state = radio.beginFSK(869.5);
...
```
Only then I had a new problem. When transmitting hello world I got a timeout. I checked the Dio mapping and irqFlags2 registers. The radio wass enabled (I could see things happening on my Spectrum analyzer) but I received a timeout:

> [SX1262] Timed out while transmitting!

Check the comments for a potential fix.

**Reply by jgromes** [reply]
Are you sure the pins are correct? The name `LORA_IRQ_DUMB` seems to suggest it might be a dummy constant.

**Reply by engeen-nl** [reply]
LORA_IRQ_DUMB is the chip select pin. I noticed though after rebooting the board it wouldn't detect the chip again. Running the reset procedure from sandeepmistry's library made it work again. So updating my answer... 

I continued with altered version of sandeepmistry's library and there I found out that I had to change RegFifoTresh (0x35 see page 101 of datasheet). Setting TxStartCondition to 0 (means using FifoLevel) and setting the FifoLevel to a sane value (default is 16, but I was sending only 7 bytes). I don't understand completely why it didn't start sending with TxStartCondition 1 because I did see that FifoEmpty was low when I started transmitting. I receive messages now on the other end. Not using the interrupt pin but polling RegIrqFlags2.

So I'm pretty sure RadioLib can work with the MKR 1310, just with some tweaking. (i.e. 
`#if defined(ARDUINO_SAMD_MKRWAN1300) || defined(ARDUINO_SAMD_MKRWAN1310)`)
